internal/core/metadata/application/device.go

```diff
@@ -165,6 +189,14 @@ func PatchDevice(dto dtos.UpdateDevice, ctx context.Context, dic *di.Container)
 		go updateDeviceCallback(ctx, dic, oldServiceName, device)
 	}
 	go updateDeviceCallback(ctx, dic, device.ServiceName, device)
+
+	go func() {
+		if err := publishDeviceSystemEvent(common.DeviceSystemEventActionUpdate, device, ctx, lc, dic); err != nil {
+			// Don't fail request if couldn't publish, just log it. Error message already has context.
+			lc.Error(err.Error())
```

Comment:
log could add more context here:
`lc.Errorf("issue happens when publishing the device update system event: %v", err)`

Reply:
That is already added in the function. I have it returning error in order to validate in unit tests. Think it is better to just have the function log the error.

Reply:
I have reworked the error handling so this is no longer needed.